internal/gatewayapi/validate.go

```diff
@@ -333,7 +335,12 @@ func (t *Translator) validateTLSConfiguration(listener *ListenerContext, resourc
 				break
 			}
 
-			err := validateTLSSecretData(secret)
+			var hostname string
+			if listener.Hostname != nil {
+				hostname = string(*listener.Hostname)
+			}
+
+			err := validateTLSSecretsData(secrets, hostname)
```

Comment:
@cnvergence you are passing an empty string when `listener.Hostname` is nil , but you ideally want to pass `*`, this might be why CI is failing :)
https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.Listener

Reply:
It was the issue with the wildcard certificate, it seems that x509.verifyHostname is not working with those certificates.
I have removed usage of it.